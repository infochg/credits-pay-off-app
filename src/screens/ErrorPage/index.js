import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Button, Container, Typography } from '@material-ui/core';
import ErrorBoundary from '../../containers/ErrorBoundary';
import errorImg from '../../assets/img/404.svg';

const useStyles = makeStyles(theme => ({
  contentWrapper: {
    display: 'flex',
    alignItems: 'center'
  },
  content: {
    flexBasis: '50%',
    '& a': {
      textDecoration: 'none'
    }
  },
  h1: {
    fontSize: '52px',
    fontWeight: '900'
  },
  text: {
    fontSize: '26px',
    color: theme.palette.text.secondary,
    margin: '30px 0'
  },
  imageWrapper: {
    flexBasis: '50%',
    '& img': {
      maxWidth: '100%'
    }
  },
  btn: {
    borderRadius: '30px',
    paddingLeft: '40px',
    paddingRight: '40px'
  },
  [theme.breakpoints.down('sm')]: {
    contentWrapper: {
      flexDirection: 'column'
    },
    content: {
      order: '2',
      flexBasis: '100%',
      textAlign: 'center'
    },
    imageWrapper: {
      order: '1',
      flexBasis: '100%'
    }
  },
  [theme.breakpoints.down('xs')]: {
    h1: {
      fontSize: '32px'
    },
    text: {
      fontSize: '18px',
      margin: '20px 0'
    }
  }
}));

function ErrorPage() {
  const classes = useStyles();

  try {
    return (
      <Container className={classes.contentWrapper}>
        <div className={classes.content}>
          <Typography className={classes.h1}>Something is wrong</Typography>
          <Typography className={classes.text}>
            The page you are looking for was moved, removed,renamed or might
            never existed
          </Typography>
          <a href="/">
            <Button
              variant="contained"
              color="primary"
              disableElevation
              className={classes.btn}
            >
              Go home
            </Button>
          </a>
        </div>
        <div className={classes.imageWrapper}>
          <img src={errorImg} alt="" />
        </div>
      </Container>
    );
  } catch (e) {
    return <ErrorBoundary error={e} />;
  }
}

export default ErrorPage;
