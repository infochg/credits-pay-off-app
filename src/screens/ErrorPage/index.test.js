import React from 'react';
import { render } from 'enzyme';
import ErrorPage from './index';

describe('<ErrorPage />', () => {
  const wrapper = render(<ErrorPage />);

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
