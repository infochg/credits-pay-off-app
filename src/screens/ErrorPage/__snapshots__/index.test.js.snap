// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage /> should match the snapshot 1`] = `"<div class=\\"makeStyles-content-2\\"><p class=\\"MuiTypography-root makeStyles-h1-3 MuiTypography-body1\\">Something is wrong</p><p class=\\"MuiTypography-root makeStyles-text-4 MuiTypography-body1\\">The page you are looking for was moved, removed,renamed or might never existed</p><a href=\\"/\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-6 MuiButton-containedPrimary MuiButton-disableElevation\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Go home</span></button></a></div><div class=\\"makeStyles-imageWrapper-5\\"><img src=\\"404.svg\\" alt></div>"`;
